### PR TITLE
Generate blueprint unit tests

### PR DESCRIPTION
**Pull request by kellyselden**
* additional unit tests for 100% coverage
* moved blueprint printing logic from generate command to blueprint model
* code cleanup/refactor Blueprint.printBasicHelp

This could use a second pair of eyes, to make sure someone agrees that blueprint option printing logic should be in the blueprint.

### CODE REVIEW

File: lib/utilities/printable-properties.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+];
+var blueprintProperties = [
+  'name',
+  'description',
+  'availableOptions',
+  'anonymousOptions',
+  'overridden'
+];
+
+function forEachProperty(properties, forEach, context) {
```
Comment (stefanpenner) [inline lib/utilities/printable-properties.js]: is this actually forEachWithProperty ?

**Comment by stefanpenner** [comment]
obviously, this is lots of (existing crappy) code so i may have missed something. But this feels like the right direction.

**Comment by stefanpenner** [comment]
Did another pass +1 from me (with the exception of my 1 comment).

File: lib/utilities/printable-properties.js (hunk 1 above)
Comment (kellyselden) [inline lib/utilities/printable-properties.js]: I don't recognize that keyword. Google didn't help. Can you explain?

Comment (stefanpenner) [inline lib/utilities/printable-properties.js]: I think it is a better name for the function you have already defined.

It appears, we will forEach only members with a specific property, hence the naming recommendation of `forEachWithProperty`

File: lib/utilities/printable-properties.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  return properties.filter(function(key) {
+    return this[key] !== undefined;
```
Comment (stefanpenner) [inline lib/utilities/printable-properties.js]: should this actually be `key` in this` ?

**Comment by kellyselden** [comment]
made the suggested change of `forEachWithProperty`

**Comment by kellyselden** [comment]
@homu r+

**Pull request merged**